src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int RATE_INDEX = 2;
+    private static final int TIME_INDEX = 3;
+    private static final String DASH = " - ";
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        StringBuilder result = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(DASH).append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            result.append(System.lineSeparator());
+            result.append(names[i])
+                    .append(DASH);
+            LocalDate localDate = LocalDate.parse(dateFrom, dateTimeFormatter);
+            LocalDate localDate1 = LocalDate.parse(dateTo, dateTimeFormatter);
```

Comment:
do not use such naming, how to guess what it stores?
```suggestion
            LocalDate dateFrom = LocalDate.parse(dateFrom, dateTimeFormatter);
            LocalDate dateTo = LocalDate.parse(dateTo, dateTimeFormatter);
```